Sources/FuzzilliCli/Profiles/XSProfile.swift

```diff
@@ -14,12 +14,379 @@
+        let codePoint = UInt32.random(in: 0..<0x10FFFF)
+        // ignore surrogate pair code points
+        if !((0xD800 <= codePoint) && (codePoint < 0xE000)) {
+            s += String(Unicode.Scalar(codePoint)!)
+        }
+    }
+    b.loadString(s)
+}
+
+fileprivate let HexGenerator = CodeGenerator("HexGenerator") { b in
+    let hexValues = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f", "A", "B", "C", "D", "E", "F"]
+
+    let Uint8Array = b.loadBuiltin("Uint8Array")
+
+    withEqualProbability({
+            var s = ""
+            for _ in 0..<Int.random(in: 1...10) {
+                s += chooseUniform(from: hexValues)
+                s += chooseUniform(from: hexValues)
+            }
+            let hex = b.loadString(s)
+
+            if probability(0.5) {
+                b.callMethod("fromHex", on: Uint8Array, withArgs: [hex])
+            } else {
+                let target = b.construct(Uint8Array, withArgs: [b.loadInt(Int64.random(in: 0...0x100))])
+                b.callMethod("setFromHex", on: target, withArgs: [hex])
+            }
+        }, {
+            var values = [Variable]()
+            for _ in 0..<Int.random(in: 1...20) {
+                values.append(b.loadInt(Int64.random(in: 0...0xFF)))
+            }
+
+            let bytes = b.callMethod("of", on: Uint8Array, withArgs: values)
+            b.callMethod("toHex", on: bytes, withArgs: [])
+        }
+    )
+}
+
+fileprivate let Base64Generator = CodeGenerator("Base64Generator") { b in
+    let base64Alphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "+", "/"]
+    let base64URLAlphabet = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "-", "_"]
+
+    let Uint8Array = b.loadBuiltin("Uint8Array")
+
+    withEqualProbability({
+            var options = [String: Variable]()
+            var alphabet = chooseUniform(from: [base64Alphabet, base64URLAlphabet])
+
+            options["alphabet"] = b.loadString((alphabet == base64Alphabet) ? "base64" : "base64url")
+            options["lastChunkHandling"] = b.loadString(
+                chooseUniform(
+                    from: ["loose", "strict", "stop-before-partial"]
+                )
+            )
+
+            var s = ""
+            for _ in 0..<Int.random(in: 1...32) * 4 {
+                s += chooseUniform(from: alphabet)
+            }
+
+            // extend by 0, 1, or 2 bytes
+            switch (Int.random(in: 0...3)) {
```

Comment:
Hm alternatively how about this idea:
We generate a random byte string, then Base64 encode that in swift and then load that as a string? :)
Then we can drop the variables and all of this?
```
var randomDataLength = UInt8.random(in: UInt8.min...UInt8.max)
var randomData = Data(capacity: randomDataLength)
for i in 0..<randomDataLength {
    randomData[i] = Uint8.random(in: UInt8.min...Uint8.max)
}

let base64 = b.loadString(randomData.base64EncodedString())
```
WDYT? 

https://developer.apple.com/documentation/foundation/data/2142853-base64encodedstring

Reply:
Nice idea. However, it would eliminate the tests for the "base64url" alphabet (Swift's Base64 encoder doesn't seem to have that [option](https://developer.apple.com/documentation/foundation/nsdata/base64encodingoptions)). I think I'd like to keep the Base64 generation: it isn't that much code and it makes it straightforward to add other options later like whitespace and invalid inputs.